mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -1525,3 +1524,24 @@ func clusterConnectionStringsSchema() *schema.Schema {
 		},
 	}
 }
+
+func isEqualProviderAutoScalingMinInstanceSize(k, old, newStr string, d *schema.ResourceData) bool {
+	canScaleDown, _ := d.GetOk("auto_scaling_compute_scale_down_enabled")
+	canScaleUp, _ := d.GetOk("auto_scaling_compute_enabled")
+	if canScaleDown != nil && canScaleUp != nil && canScaleUp.(bool) && canScaleDown.(bool) {
```

Comment:
[q] Is `canScaleUp.(bool) && canScaleDown.(bool)` really needed? is it not enough to check that is not `nil`?

Reply:
[q] why aren’t we checking that `canScaleDown == true || canScaleUp== true`?

Reply:
I'm checking if there's not nil (the result of the `GetOk` is an interface) before try to evaluate the variables,  and also when we evaluate the var `canScaleDown`, the name has the behavior implicit and it means that you `can` only have 2 possible boolean values

makes more sense to evaluate `if isTrueVariable then` than `if IsTrueVariable == true then`